src/GeositeFramework/js/Map.js

```diff
@@ -283,20 +293,21 @@ require(['use!Geosite',
             view.$legendEl.empty()
                 .append.apply(view.$legendEl, legendNuggets);
 
-            // compare the total number of visible layers on the map to the number
+            // Compare the total number of visible layers on the map to the number
             // of legendNuggets to determine if the legend has gotten info for all
             // visible layers. If not, recur with a delay.
-            // this is a hack, because sometimes when loading from browser cache,
+            //
+            // This is a hack, because sometimes when loading from browser cache,
             // layerAdd events don't fire.
-            function countVisibleLayers(count, layerInfo) {
-                var vl = layerInfo.layer.visibleLayers,
-                    shouldAdd = !(vl.length === 1 && vl[0] === -1);
-                return count += shouldAdd ? vl.length : 0;
-            }
-            var totalVisibleLayers =_.reduce(layerInfos, countVisibleLayers, 0);
 
-            if (totalVisibleLayers !== legendNuggets.length) {
-                _.delay(updateLegend, 1000);
+            // Get all the layers on the map, and filter out graphic layers (i.e. subregion boundaries).
+            var totalVisibleLayers =_.filter(esriMap.getLayersVisibleAtScale(), function(layer) {
+                return !layer.graphics;
+            });
+
+            // Subtract one for the base layer.
+            if ((totalVisibleLayers.length - 1) !== nuggetCount) {
+                _.delay(updateLegend, 250);
```

Comment:
That was in there, I just shortened the time. Here's the relevant comment:

```
// Compare the total number of visible layers on the map to the number
// of legendNuggets to determine if the legend has gotten info for all
// visible layers. If not, recur with a delay.
//
// This is a hack, because sometimes when loading from browser cache,
// layerAdd events don't fire.
```